esp-hal/src/lcd_cam/lcd/i8080.rs

```diff
@@ -467,9 +363,25 @@ impl<'d, CH: DmaChannel, DM: Mode> I8080<'d, CH, DM> {
             .lcd_2byte_en()
             .bit(is_2byte_mode)
         });
-    }
 
-    fn start_send(&mut self) {
+        // Use continous mode for DMA. FROM the S3 TRM:
+        // > In a continuous output, LCD module keeps sending data till:
+        // > i. LCD_CAM_LCD_START is cleared;
+        // > ii. or LCD_CAM_LCD_RESET is set;
+        // > iii. or all the data in GDMA is sent out.
+        self.lcd_cam
+            .lcd_user()
+            .modify(|_, w| w.lcd_always_out_en().set_bit().lcd_dout().set_bit());
+
+        let result = unsafe {
+            self.tx_channel
+                .prepare_transfer(DmaPeripheral::LcdCam, &mut data)
+        }
+        .and_then(|_| self.tx_channel.start_transfer());
+        if let Err(err) = result {
```

Comment:
How annoying would it be to move the error handling into the transaction? It would be nice if we could make `send` infallible and let the transaction be the only point where the user meets the error.

Reply:
Omg yes! I'm planning to do this. There's no reason to be checking for DMA errors this early, or at least the checking can wait until later in `is_done`\`wait`.

I'll do this for both I8080 and SPI in my follow up PR.

Reply:
Oh no wait, I've misunderstood you slightly.

Due to how the Channel implementation works, I could technically just ignore the `DmaError` returned by `prepare` and `start`, and the user will eventually encounter it when calling `wait`. (I had this initially but I was worried I'd be told off for ignoring the error)

I could store the error in the transfer object but that causes problems in my follow up PRs so I'd rather not go that path.

Reply:
I'm curious to see what you're cooking here